faf-commons-data/src/main/java/com/faforever/commons/replay/ReplaySemantics.java

```diff
@@ -0,0 +1,152 @@
+package com.faforever.commons.replay;
+
+import com.faforever.commons.replay.body.Event;
+import com.faforever.commons.replay.header.Source;
+import com.faforever.commons.replay.shared.LuaData;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ReplaySemantics {
+
+  public static Duration tickToDuration(int tick) {
+    return Duration.ofSeconds(tick / 10);
+  }
+
+  /**
+   * Registers the events by attaching a tick and a source to them.
+   * @param sources
+   * @param events
+   * @return
+   */
+  public static List<RegisteredEvent> registerEvents(List<Source> sources, List<Event> events) {
+    final AtomicInteger tick = new AtomicInteger(0);
+    final AtomicInteger clientId = new AtomicInteger(-1);
+
+    return events.stream().map((event) -> switch (event) {
+      case Event.Advance e -> {
+        tick.addAndGet(e.ticksToAdvance());
+        yield null;
+      }
+
+      case Event.SetCommandSource e -> {
+        clientId.set(e.playerIndex());
+        yield null;
+      }
```

Comment:
You can use the full pattern matching form here so Event.Advance(int ticksToAdvance)

Reply:
Processed in https://github.com/FAForever/faf-java-commons/pull/143/commits/706ad773caea6a74e2babd6b0b2dbfeea4324bdd